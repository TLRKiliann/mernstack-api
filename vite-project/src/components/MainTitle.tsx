import {Component} from 'react'
import bamImg from '../assets/logo/bam-bg.png'
import './styleComponents/MainTitle.scss'

interface HeaderProps {
  textHeader: string
  secondTextHeader: string
}

export default class Header extends Component<HeaderProps> {

  /*myColor = {
    width: '40%',
    height: 'auto',
    margin: 'auto',
    marginTop: '100px',
    textAlign: 'center',
    fontFamily: "Junction, serif",
    color: "whitesmoke",
    textShadow: "4px 3px 1px #383838",
    backgroundColor: "rgba(45, 85, 255, 0.4)",
    border: "1mm outset rgba(180, 85, 255, 0.2)",
    borderRadius: '15px'
  }*/
  render() {
    return(
      <div className="maintitle">
        <div className="div--bamimg">
          <img
            src={bamImg}
            width="100%"
            height="100%"
            className="bam--img"
            alt={bamImg}
          />
        </div>
        <h1 className="main--h1">Chat-Society</h1>
        <h2 className="main--h2">{this.props.textHeader}</h2>
        <h4 className="main--h4">{this.props.secondTextHeader}</h4>
      </div>
    )
  }
}