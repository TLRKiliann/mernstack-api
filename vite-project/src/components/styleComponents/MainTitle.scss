$darkorchid: #9932cc;
$pink: #ffc0cb;
$orange: #ffa500;
$orangered: #ff4500;
$font_acme: 'Acme', sans-serif;

.maintitle {
  width: 40%;
  height: 150px;
  padding-top: 60px;
  padding-bottom: 60px;
  margin: auto;
  margin-top: 140px;
  text-align: center;
  font-family: $font_acme;
  text-shadow: -3px 3px 2px $darkorchid,
    -4px 4px 3px #000;
  background: linear-gradient(180deg, rgba(112, 43, 183, 0.2),
    rgba(112, 43, 183, 0.4));
  border: none;
  outline: none;
  border-radius: 12px;
  box-shadow: 0px 0px 7px rgba(112, 43, 183, 0.4);  
  .div--bamimg {
    position: absolute;
    width: 42%;
    height: 250px;
    margin-top: -50px;
    margin-left: -10px;
    opacity: 0.4;

    .bam--img {
      width: 90%;
      margin: auto;
    }
  }
  .main--h1 {
    margin: auto;
    margin-top: -30px;
    margin-bottom: 0px;

    font-size: 5.4rem;
    text-shadow: -3px 3px 2px $darkorchid,
      -10px 10px 7px #424242;
    transform: rotateZ(-10deg);
    color: $pink;
  }
  .main--h2 {
    margin: auto;
    margin-top: -15px;
    margin-bottom: 15px;
    margin-left: 40px;
    font-size: 3.4rem;
    color: sandybrown;
    text-shadow: -3px 3px 2px $orangered,
      -8px 8px 5px #424242;
    transform: rotateZ(-10deg);
  }
  .main--h4 {
    margin: auto;
    margin-top: -10px;
    margin-bottom: 10px;
    margin-left: 60px;
    font-size: 1.1rem;
    text-shadow: -2px 2px 2px $darkorchid,
      -4px 4px 3px #383838;
    transform: rotateZ(-10deg);
    color: $pink;
  }
}

@media screen and (min-width: 360px) and (max-width: 900px) {
  .maintitle {
    width: 360px;
    height: 180px;
    padding-top: 20px;
    padding-bottom: 20px;
    margin: auto;
    margin-top: 140px;
    border: none;
    outline: none;
    .div--bamimg {
      margin: auto;
      margin-top: -15px;
      width: 360px;
      height: 200px;
      display: flex;
      align-items: center;
      opacity: 0.4;
    }
    .main--h1 {
      margin-top: 0px;
      font-size: 3.8rem;
    }
    .main--h2 {
      margin-top: -10px;
      font-size: 2.2rem;
    }
    .main--h4 {
      font-size: 0.9rem;
    }
  }
}


@media screen and (min-width: 900px) and (max-width: 1068px) {
  .maintitle {
    width: 360px;
    height: 160px;
    padding-top: 20px;
    margin: auto;
    margin-top: 140px;
    background: linear-gradient(180deg, rgba(112, 43, 183, 0.2),
      rgba(112, 43, 183, 0.4));
    .div--bamimg {
      opacity: 0.4;
      .bam--img {
        margin-top: 50px;
        margin-left: -20px;
        width: 340px;
        height: 200px;
      }
    }
    .main--h1 {
      margin-top: 20px;
      font-size: 3.8rem;
    }
    .main--h2 {
      margin-top: -10px;
      font-size: 2.8rem;
    }
    .main--h4 {
      font-size: 1.0rem;
    }
  }
}