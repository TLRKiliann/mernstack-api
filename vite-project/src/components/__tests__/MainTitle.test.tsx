import React from "react";
import { screen, render } from '@testing-library/react';
import { test, expect, vi } from 'vitest'
import { create } from 'react-test-renderer'
import '@testing-library/jest-dom'
import MainTitle from "../MainTitle.tsx";

test('MainTitle getByText', () => {
  render(<MainTitle />);
  const linkElement = screen.getByText(/Chat-Society/i);
  expect(linkElement).toBeInTheDocument();
})

test('MatchSnapShot test MainTitle', () => {
  const treeMainTitle = create(<MainTitle />)
  expect(treeMainTitle.toJSON()).toMatchSnapshot()
})