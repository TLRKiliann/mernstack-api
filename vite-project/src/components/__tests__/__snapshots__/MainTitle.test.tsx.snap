// Vitest Snapshot v1

exports[`MatchSnapShot test MainTitle 1`] = `
<div
  className="maintitle"
>
  <h1
    className="main--h1"
  >
    Chat-Society
  </h1>
  <h2
    className="main--h2"
  />
  <h4
    className="main--h4"
  />
</div>
`;
